app/src/hooks/useXdaiBridge.tsx

```diff
@@ -0,0 +1,146 @@
+import { Zero } from 'ethers/constants'
+import { BigNumber, bigNumberify } from 'ethers/utils'
+import { useEffect, useState } from 'react'
+
+import { XdaiService } from '../services'
+import { formatBigNumber } from '../util/tools'
+
+import { useConnectedCPKContext } from './connectedCpk'
+import { useConnectedWeb3Context } from './connectedWeb3'
+
+export enum State {
+  idle,
+  waitingConfirmation,
+  transactionSubmitted,
+  transactionConfirmed,
+  error,
+}
+interface Prop {
+  transactionHash: string
+  transferFunction: any
+  fetchUnclaimedAssets: any
+  daiBalance: BigNumber
+  xDaiBalance: BigNumber
+  transactionStep: State
+  claimLatestToken: any
+  unclaimedAmount: BigNumber
+  claimState: boolean
+  isClaimStateTransaction: boolean
+}
+
+export const useXdaiBridge = (amount?: BigNumber): Prop => {
+  const [transactionStep, setTransactionStep] = useState<State>(State.idle)
+  const { account, library: provider, networkId } = useConnectedWeb3Context()
+  const [xDaiBalance, setXdaiBalance] = useState<BigNumber>(Zero)
+  const [daiBalance, setDaiBalance] = useState<BigNumber>(Zero)
+  const [isClaimStateTransaction, setIsClaimStateTransaction] = useState<boolean>(false)
+  const [unclaimedAmount, setUnclaimedAmount] = useState<BigNumber>(Zero)
+  const [claimState, setClaimState] = useState<boolean>(false)
+  const [transactionHash, setTransactionHash] = useState<string>('')
+  const cpk = useConnectedCPKContext()
+
+  const transferFunction = async () => {
+    try {
+      if (!cpk || !amount) return
+      if (networkId === 1) {
+        setTransactionStep(State.waitingConfirmation)
+
+        const transaction = await cpk.sendDaiToBridge(amount)
+
+        setTransactionHash(transaction.hash)
+        setTransactionStep(State.transactionSubmitted)
+
+        await cpk.waitForTransaction(transaction, true)
+        setTransactionStep(State.transactionConfirmed)
+      } else {
+        const amountInFloat = formatBigNumber(amount, 18)
+
+        if (parseInt(amountInFloat) < 10) {
```

Comment:
Like you said in your comment above, we probably need to mention somewhere that the min transfer is 10 Dai. I forgot about this and got confused when I couldn't transfer 😄

Reply:
yeah we would probably need to error modal state! @pimato What do you think...